book/platform/README.md

```diff
@@ -0,0 +1,473 @@
+
+<!-- $MDX file=examples/correct/hello-world/hello.opam -->
+```
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "A short description of the project"
+description: "A short description of the project"
+maintainer: ["Your name"]
+authors: ["Your name"]
+license: "ISC"
+homepage: "https://github.com/username/hello"
+doc: "https://username.github.io/hello/"
+bug-reports: "https://github.com/username/hello/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune"
+  "alcotest" {with-test}
+  "odoc" {with-doc}
+]
+build: [
+  ["dune" "subst"] {pinned}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/username/hello.git"
+```
+
+The fields in here all represent project metadata ranging
+from textual descriptions, to project URLs, to other opam package dependencies.
+A collection of `opam` files can be stored in an
+*opam repository* to create a package database, with a central one for the OCaml
+ecosystem available at <https://github.com/ocaml/opam-repository>.
+The official (but not exclusive) tool used for manipulating `opam` files is the
+eponymous [opam package manager](https://opam.ocaml.org) that we've been using
+throughout this book.
+
+### Generating project metadata from dune
+
+You don't need to write the opam file by hand -- instead, we can use
+define our project metadata using the dune build system and have it autogenerated for us.
+We've already been using the `dune` build tool from back in [Files Modules And Programs](files-modules-and-programs.html#files-modules-and-programs){data-type=xref}. We're now going to look at how to configure your project for use with dune.
+
+The root directory of an OCaml project built by dune has a `dune-project` file that defines the project metadata. In our hello world example, it starts with:
+
+```scheme
+(lang dune 2.0)
+```
+
+The line above is the version of the syntax used in your build files, and _not_
+the actual version of the `dune` binary.  One of the nicest features of dune is that
+it is forwards-compatible with older metadata. By defining the version of
+the dune language that you are currently using, _future_ versions of dune will do their best to emulate the current behaviour until you chose to upgrade your project.
+
+The rest of the `dune-project` file defines other useful project metadata:
+
+```scheme
+(name hello)
+(documentation "https://username.github.io/hello/")
+(source (github username/hello))
+(license ISC)
+(authors "Your name")
+(maintainers "Your name")
+(generate_opam_files true)
+```
+
+The fields here should look familiar -- they were also present in the `hello.opam` file
+above.  That's because dune can generate the opam packaging metadata files for you and
+avoid duplication.  Go ahead and edit the metadata above, and then build the project with:
+
+```skip
+$ dune build
```

Comment:
Could be `dune build hello.opam`? This serves two goals:
- Show users they can build a specific target with `dune build`
- Show them they can generate the opam file even if the project does not compile